File: AulasSpring/Portal-Aula/src/app/dto/internal/ResponseEnvelope.ts
export class ResponseEnvelope {

  private _object : any | undefined;

  private _success : boolean | undefined;

  private _message : string | undefined;


  get object(): any {
    return this._object;
  }

  set object(value: any) {
    this._object = value;
  }

  get success(): boolean | undefined {
    return this._success;
  }

  set success(value: boolean | undefined) {
    this._success = value;
  }

  get message(): string | undefined {
    return this._message;
  }

  set message(value: string | undefined) {
    this._message = value;
  }
}

File: AulasSpring/AulaService/src/main/java/com/aulas/maven/aulaservice/controllers/internal/ResponseEnvelope.java
package com.aulas.maven.aulaservice.controllers.internal;

import java.io.Serializable;

public class ResponseEnvelope<T> implements Serializable {

    private T object;

    private boolean success;

    private String message;

    public T getObject() {
        return object;
    }

    public ResponseEnvelope<T> setObject(T object) {
        this.object = object;
        return this;
    }

    public boolean isSuccess() {
        return success;
    }

    public ResponseEnvelope<T> setSuccess(boolean success) {
        this.success = success;
        return this;
    }

    public String getMessage() {
        return message;
    }

    public ResponseEnvelope<T> setMessage(String message) {
        this.message = message;
        return this;
    }
}
